BetaCameras/ifm/O3D3xx.cs

```diff
@@ -153,44 +234,39 @@ private ParamDesc<String> ImageOutputPortDesc
                 return res;
             }
         }
+        #endregion
 
 #if !NETSTANDARD2_0
         public override System.Drawing.Icon CameraIcon { get => Properties.Resources.IfmIcon; }
 #endif
         #endregion
 
         #region Private Fields
-        private Socket clientSocket;
-        private bool configurationMode = false;
-        private int triggeredMode = 1;
-        private int frequencyChannel = 0;
-        private int framerate = 25;
-        private int exposureTime = 1234;
-        private int applicationId;
-        private int width;
-        private int height;
-
-        private FloatCameraImage latestAmplitudeImage;
-        private FloatCameraImage latestDistanceImage;
-        private Point3fCameraImage latestPoint3fImage;
-        private FloatCameraImage latestZImage;
-        private ByteCameraImage latestRawConfidenceImage;
-
-        private FloatCameraImage amplitudeImage;
-        private FloatCameraImage distanceImage;
-        private Point3fCameraImage point3fImage;
-        private FloatCameraImage zImage;
-        private ByteCameraImage rawConfidenceImage;
-
-        private ISession session;
-        private IDevice device;
-        private IAppImager appImager;
-        private IApp app;
-        private IEdit edit;
-        private IEditDevice editeDevice;
-        private IServer server;
-
-        private string serverUrl;
+        private Thread _updateThread;
+        private bool _cancelUpdateThread = false;
```

Comment:
Please use a `CancellationToken` instead of a boolean. One day, this flag might need to be passed to other asynchronous operations.